statefun_tasks/_flink_tasks.py

```diff
@@ -21,7 +22,7 @@ def _create_task_exception(task_request, ex, retry=False):
         id=_gen_id(),
         correlation_id=task_request.id,
         type=f'{task_request.type}.error',
-        exception_type=type(ex).__name__,
+        exception_type=_type_name(ex),
         exception_message=str(ex),
         stacktrace=tb.format_exc(),
```

Comment:
I know this isn't part of the review but I think that the stacktrace should be an array of file line references rather than a string. There should be a separate cause field which will be set if the exception was raised from another exception and could recursively contain further exceptions. If you think that makes sense, I can make the change.

Reply:
Will do in later release